src/modules/Elsa.ProtoActor/Services/ProtoActorWorkflowRuntime.cs

```diff
@@ -249,10 +249,51 @@ public async Task<WorkflowExecutionResult> ExecuteWorkflowAsync(WorkflowMatch ma
     }
 
     /// <inheritdoc />
-    public async Task CancelWorkflowAsync(string workflowInstanceId, CancellationToken cancellationToken)
+    public async Task<CancellationResult> CancelWorkflowAsync(string workflowInstanceId, CancellationToken cancellationToken)
     {
+        var filter = new WorkflowInstanceFilter
+        {
+            Id = workflowInstanceId
+        };
+
+        var instance = await _workflowInstanceStore.FindAsync(filter);
+        if (instance is null)
+            return new CancellationResult(false, FailureReason.NotFound);
+        
         var client = _cluster.GetNamedWorkflowGrain(workflowInstanceId);
-        await client.Cancel(cancellationToken);
+        var result = await client.Cancel(cancellationToken);
+        return new CancellationResult(result?.Result ?? false);
+    }
+
+    public async Task<int> CancelWorkflowAsync(IEnumerable<string> workflowInstanceIds, CancellationToken cancellationToken = default)
```

Comment:
We should probably rename this to `CancelWorkflowsAsync` (plural), given that it supports the cancellation of a list of them.